e2e/test_operator/operator_test.go

```diff
@@ -370,13 +371,28 @@ var _ = Describe("Operator", Label("e2e", "pr"), func() {
 		// fields that we expect are actually set.
 	})
 
-	When("replacing a Pod", func() {
+	When("replacing a coordinator Pod", func() {
 		var replacedPod corev1.Pod
 		var useLocalitiesForExclusion bool
 
 		JustBeforeEach(func() {
-			initialPods := fdbCluster.GetStatelessPods()
-			replacedPod = factory.RandomPickOnePod(initialPods.Items)
+			initialPods := fdbCluster.GetLogPods()
+			coordinators := fdbstatus.GetCoordinatorsFromStatus(fdbCluster.GetStatus())
+
+			for _, pod := range initialPods.Items {
+				_, isCoordinator := coordinators[string(fixtures.GetProcessGroupID(pod))]
+				if isCoordinator {
+					replacedPod = pod
+					break
+				}
+			}
+
+			// In case that none of the log processes are coordinators
+			if replacedPod.Name == "" {
+				replacedPod = factory.RandomPickOnePod(initialPods.Items)
```

Comment:
is the intention to prefer coordinator logs and then just pick anything else otherwise? why prefer logs that are coordinators in the first place - just to try to add some coverage on fun cases?

Reply:
The idea is to prefer coordinators to make sure the operator handles the case with replacing a coordinator well (with the assumption that if the operator handles the coordinator replacement well it also handles the normal replacement well). If no log processes are coordinators the test suite will randomly pick a log process.